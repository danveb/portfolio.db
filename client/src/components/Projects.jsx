import { projects } from "../constants/projects";
import "../styles/Projects.scss"; 

const Projects = () => {
    return (
        <div id="projects" className="projects">
            <div className="projects__wrapper">
                <div className="projects__main">
                    <div className="projects__title">
                        <h3>Projects<span>.</span></h3>
                    </div>
                    <div className="projects__cards">
                        {projects.map((project) => (
                            <div className="card__container" key={project.id}>
                                <div className="image__container">
                                    <img src={project.img} alt="portfolio project showcase" style={{backgroundColor: `${project.backgroundColor}`}}/>
                                </div>
                                <div className="card__content">
                                    <div className="card__title">
                                        <h1>{project.title}</h1>
                                    </div>
                                    <div className="card__body">
                                        <p>{project.description}</p>
                                    </div>
                                    {project.buttons.map((item) => (
                                        <div className="btn" key={item}>
                                            <button>{item}</button>
                                        </div>
                                    ))} 
                                    <div className="card__footer">
                                        <a href={project.liveSite} target={project.target} rel={project.rel}>{project.liveSite !== "" ? "Live Site" : ""}</a>
                                        <a href={project.githubRepo} target={project.target} rel={project.rel}>{project.githubRepo !== "" ? "GitHub Repo" : ""}</a>
                                    </div>
                                </div>
                            </div>
                        ))}
                    </div>
                </div>
            </div>
        </div>
    )
}

export default Projects