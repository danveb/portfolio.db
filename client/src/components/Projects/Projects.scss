.projects {
    display: flex; 
    justify-content: center; 
    align-items: center; 
    flex-wrap: wrap; 
    background-color: var(--alice-blue); 

    .card-container {
        width: 500px; 
        margin: 30px; 
        overflow: hidden; 
        box-shadow: 0 0 15px -5px; 
        transition: all 0.3s ease-in; 

        &:hover {
            transform: scale(1.05); 
            box-shadow: 0 0 15px -5px; 
        }

        .image-container {
            
            img {
                height: 300px; 
                width: 100%; 
                object-fit: contain; 
                overflow: hidden; 
            }
        }

        .card-content {
            margin: 20px; 
            
            .card-title {
                margin: 10px; 
                font-family: "Playfair Display", sans-serif; 
            
                h1 {
                    font-size: 26px; 
                    font-family: inherit; 
                    font-weight: 400; 
                }
            }

            .card-body {
                margin: 10px; 
                font-family: "Playfair Display", sans-serif; 

                p {
                    font-size: 14px; 
                    font-family: inherit; 
                }
            }

            .btn {
                margin: 10px; 
                display: inline-block; 
                font-family: "Playfair Display", sans-serif; 

                button {
                    background-color: var(--languid-lavender); 
                    border: 1px solid transparent; 
                    padding: 5px 7px; 
                    text-transform: uppercase; 
                    font-size: 10px; 
                    font-family: inherit; 
                }
            }

            .card-footer {
                margin: 10px; 
                display: flex; 
                flex-direction: column; 
                font-family: "Playfair Display", sans-serif; 

                a {
                    padding: 3px; 
                    color: inherit; 
                    font-family: inherit; 
                    text-decoration: none; 
                    transition: all 0.5s ease-out; 

                    &:hover {
                        color: var(--bottle-green); 
                        text-decoration: underline; 
                    }
                }
            }
        }
    }
}