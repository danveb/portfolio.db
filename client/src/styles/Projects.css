.projects {
  width: 100%; 
  height: 100%; 
}

.projects__wrapper {
  max-width: 768px; 
  margin: 0 auto; 
  width: 100%; 
  height: 100%; 
  background-color: var(--rich-black); 
}

.projects__main {
  display: flex; 
  flex-direction: column; 
  justify-content: center; 
  gap: 20px; 
  margin: 40px 0;
}

.projects__title {
  font-family: "Mulish", sans-serif; 
  font-size: 20px; 
  color: var(--alice-blue); 
  letter-spacing: 0.5px; 
}

.projects__intro {
  font-family: inherit; 
  font-size: 16px; 
  font-weight: 300; 
  color: #d4d4d4; 
  letter-spacing: 1px; 
  line-height: 1.5rem; 
}

.projects__intro p span {
  color: #fff;
  font-weight: 600; 
}

@media screen and (max-width: 1024px) {
  .projects__wrapper {
    padding: 0 20px; 
  }
}