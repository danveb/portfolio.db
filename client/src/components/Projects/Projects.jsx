import { projects } from "../../constants/projects";
import "./Projects.scss"; 

const Projects = () => {
    return (
        <div id="projects" className="projects">
            {projects.map((project) => (
                <div className="card-container" key={project.id}>
                    <div className="image-container">
                        <img src={project.img} alt="portfolio project showcase" style={{backgroundColor: `${project.backgroundColor}`}}/>
                    </div>
                    <div className="card-content">
                        <div className="card-title">
                            <h1>{project.title}</h1>
                        </div>
                        <div className="card-body">
                            <p>{project.description}</p>
                        </div>
                        {project.buttons.map((item) => (
                            <div className="btn" key={item}>
                                <button>{item}</button>
                            </div>
                        ))} 
                        <div className="card-footer">
                            <a href={project.liveCode} target={project.target} rel={project.rel}>{project.liveCode !== "" ? "Live Demo" : ""}</a>
                            <a href={project.sourceCode} target={project.target} rel={project.rel}>{project.sourceCode !== "" ? "GitHub Repo" : ""}</a>
                        </div>
                    </div>
                </div>
            ))}
        </div>
    )
}

export default Projects