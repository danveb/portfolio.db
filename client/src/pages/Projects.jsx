import { ProjectList } from "../components"; 
import "../styles/Projects.css"; 

export default function Projects() {
  return (
    <div className="projects">
      <div className="projects__wrapper">
        <section className="projects__main">
          <div className="projects__title" data-testid="title">
            <h1>my projects<span className="dot--green">.</span> ðŸš€</h1>
          </div>
          <div className="breakline"></div>
          <div className="projects__intro">
            <p>Here are some of my most loved projects: </p>
          </div>
        </section>
        <section className="projects__gallery">
          <ProjectList />
        </section>
      </div>
    </div>
  )
}