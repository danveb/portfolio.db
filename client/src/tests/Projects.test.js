import { render } from "@testing-library/react"; 
import "@testing-library/jest-dom/extend-expect";
import Projects from "../components/Projects/Projects"; 

// smoke test
it("renders without crashing", () => {
    render(<Projects />); 
}); 

// snapshot test
test("matches snapshot", () => {
    const { asFragment } = render(<Projects />); 
    expect(asFragment()).toMatchSnapshot(); 
});

// getByText
// <h1>{project.title}</h1>
test("should display the header tag", () => {
    const { getByText } = render(<Projects />); 
    const header = getByText("bobaLife"); 
    expect(header).toBeInTheDocument(); 
});

// <div id="projects" className="projects">...<div>
test("should have class projects", () => {
    const { container } = render(<Projects />); 
    const projects = container.getElementsByClassName("projects"); 
    expect(projects.length).toBe(1); 
}); 