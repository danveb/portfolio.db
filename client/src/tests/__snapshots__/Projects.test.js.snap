// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="projects"
    id="projects"
  >
    <div
      class="projects-wrapper"
    >
      <div
        class="card-container"
      >
        <div
          class="image-container"
        >
          <img
            alt="portfolio project showcase"
            src="bobaLife.png"
          />
        </div>
        <div
          class="card-content"
        >
          <div
            class="card-title"
          >
            <h1>
              bobaLife
            </h1>
          </div>
          <div
            class="card-body"
          >
            <p>
              Who enjoys sipping an ice-cold boba milk tea at any time of the day? bobaLife is an e-commerce web app designed for a fictitous boba shop. This app incorporates working cart/checkout process and integration of Stripe mobile payments.
            </p>
          </div>
          <div
            class="btn"
          >
            <button>
              React
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              Redux
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              Styled Components
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              Express
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              JWT
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              Stripe-js
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              MongoDB
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              Netlify
            </button>
          </div>
          <div
            class="card-footer"
          >
            <a
              href="https://bobalife.netlify.app"
              rel="noreferrer noopener"
              target="_blank"
            >
              Live Site
            </a>
            <a
              href="https://github.com/danveb/bobalife"
              rel="noreferrer noopener"
              target="_blank"
            >
              GitHub Repo
            </a>
          </div>
        </div>
      </div>
      <div
        class="card-container"
      >
        <div
          class="image-container"
        >
          <img
            alt="portfolio project showcase"
            src="asador.png"
          />
        </div>
        <div
          class="card-content"
        >
          <div
            class="card-title"
          >
            <h1>
              asador
            </h1>
          </div>
          <div
            class="card-body"
          >
            <p>
              Ever wondered about asado (barbeque) in Buenos Aires? Asador is all about sharing the joy of the best parrillas (steakhouses). Users are greeted with a live map with pins. Signing up for an account allows users to pin their favorite parrillas.
            </p>
          </div>
          <div
            class="btn"
          >
            <button>
              React
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              Redux
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              Sass
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              Express
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              JWT
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              MapboxGL API
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              MongoDB
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              Netlify
            </button>
          </div>
          <div
            class="card-footer"
          >
            <a
              href="https://asador.netlify.app"
              rel="noreferrer noopener"
              target="_blank"
            >
              Live Site
            </a>
            <a
              href="https://github.com/danveb/asador.ly"
              rel="noreferrer noopener"
              target="_blank"
            >
              GitHub Repo
            </a>
          </div>
        </div>
      </div>
      <div
        class="card-container"
      >
        <div
          class="image-container"
        >
          <img
            alt="portfolio project showcase"
            src="ioniq5.png"
          />
        </div>
        <div
          class="card-content"
        >
          <div
            class="card-title"
          >
            <h1>
              ioniq5 Lookup
            </h1>
          </div>
          <div
            class="card-body"
          >
            <p>
              App designed for users who want to search deliveries of 2022 Hyundai IONIQ 5 to local dealerships. User can fill in the zipcode and radius. Data is fetched from Hyundai's API and key details are displayed on a table.
            </p>
          </div>
          <div
            class="btn"
          >
            <button>
              React
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              Sass
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              MUI
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              Express
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              Hyundai USA API
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              Netlify
            </button>
          </div>
          <div
            class="card-footer"
          >
            <a
              href="https://ioniq5-lookup.netlify.app/"
              rel="noreferrer noopener"
              target="_blank"
            >
              Live Site
            </a>
            <a
              href="https://github.com/danveb/ioniq5"
              rel="noreferrer noopener"
              target="_blank"
            >
              GitHub Repo
            </a>
          </div>
        </div>
      </div>
      <div
        class="card-container"
      >
        <div
          class="image-container"
        >
          <img
            alt="portfolio project showcase"
            src="sailmasterII.png"
          />
        </div>
        <div
          class="card-content"
        >
          <div
            class="card-title"
          >
            <h1>
              Sail Master II
            </h1>
          </div>
          <div
            class="card-body"
          >
            <p>
              Sail-Master II is an app for sailors and water sportsâ€™ enthusiasts that have a common goal to share the joy in sailing. This app lets users know if sailing conditions based on current wind parameters are appropriate to enjoy a day out on the water.
            </p>
          </div>
          <div
            class="btn"
          >
            <button>
              React
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              Express
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              JWT
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              Weather API
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              PostgreSQL
            </button>
          </div>
          <div
            class="btn"
          >
            <button>
              Surge
            </button>
          </div>
          <div
            class="card-footer"
          >
            <a
              href="https://lighthouse-sailor.surge.sh/"
              rel="noreferrer noopener"
              target="_blank"
            >
              Live Site
            </a>
            <a
              href="https://github.com/danveb/sail-master-II"
              rel="noreferrer noopener"
              target="_blank"
            >
              GitHub Repo
            </a>
          </div>
        </div>
      </div>
      <div
        class="card-container"
      >
        <div
          class="image-container"
        >
          <img
            alt="portfolio project showcase"
            src="laptop1.png"
          />
        </div>
        <div
          class="card-content"
        >
          <div
            class="card-title"
          >
            <h1>
              Check my GitHub
            </h1>
          </div>
          <div
            class="card-body"
          >
            <p>
              Check my GitHub link to view more projects where I continue to build user-focused apps. Follow the link below!
            </p>
          </div>
          <div
            class="card-footer"
          >
            <a
              href=""
              rel="noreferrer noopener"
              target="_blank"
            />
            <a
              href="https://github.com/danveb"
              rel="noreferrer noopener"
              target="_blank"
            >
              GitHub Repo
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
